modules/gcve-service-networking/main.tf

```diff
@@ -0,0 +1,157 @@
+  project     = var.project_id
+  description = lookup(each.value, "description", null)
+  secondary_ip_range = [
+    for i in range(
+      length(
+        contains(
+        keys(var.secondary_ranges), each.value.subnet_name) == true
+        ? var.secondary_ranges[each.value.subnet_name]
+        : []
+    )) :
+    var.secondary_ranges[each.value.subnet_name][i]
+  ]
+
+  purpose          = lookup(each.value, "purpose", null)
+  role             = lookup(each.value, "role", null)
+  stack_type       = lookup(each.value, "stack", null)
+  ipv6_access_type = lookup(each.value, "ipv6_type", null)
+}
+
+resource "google_compute_global_address" "private_ip_alloc" {
+  for_each      = var.private_ip_allocation_details
+  project       = var.project_id
+  name          = each.key
+  address       = each.value.address
+  purpose       = var.address_purpose
+  address_type  = var.address_type
+  prefix_length = each.value.prefix_length
+  network       = var.create_network ? google_compute_network.network[0].id : data.google_compute_network.network[0].id
+}
+
+resource "google_service_networking_connection" "gcve_psa" {
+  network                 = var.create_network ? google_compute_network.network[0].id : data.google_compute_network.network[0].id
+  service                 = var.service
+  reserved_peering_ranges = [google_compute_global_address.private_ip_alloc[var.primary_private_ip_alloc].name]
+  depends_on              = [google_compute_network.network]
+}
+
+resource "google_compute_network_peering_routes_config" "peering_routes" {
+  project              = var.project_id
+  peering              = var.peering
+  network              = var.create_network ? google_compute_network.network[0].name : var.network_name
+  import_custom_routes = true
+  export_custom_routes = true
+  depends_on           = [google_service_networking_connection.gcve_psa]
+}
+
+data "google_compute_network_peering" "servicenetworking" {
+  name    = "servicenetworking-googleapis-com"
+  network = google_compute_network.network[0].self_link
+  depends_on = [
+    google_compute_network_peering_routes_config.peering_routes,
+    google_service_networking_connection.gcve_psa,
+    google_compute_global_address.private_ip_alloc
+
+  ]
+}
+
+data "google_vmwareengine_network" "gcve_network" {
+  count    = var.create_gcve_network ? 0 : 1
+  provider = google-beta
+  name     = var.gcve_network_name
+  location = var.gcve_network_location
+}
+
+resource "google_vmwareengine_network" "gcve_network" {
+  count       = var.create_gcve_network ? 1 : 0
+  provider    = google-beta
+  project     = var.project_id
+  name        = var.gcve_network_name
+  location    = var.gcve_network_location
+  type        = var.gcve_network_type
+  description = var.gcve_network_description
+}
+
+module "gcve_private_connection" {
+  source = "terraform-google-modules/gcloud/google"
+
+  platform               = "linux"
+  additional_components  = ["kubectl", "beta"]
```

Comment:
do we actually need a dependency on kubectl?

Reply:
Added Changes as per review